src/platform/packages/shared/kbn-lens-embeddable-utils/config_builder/config_builder.ts

```diff
@@ -90,30 +103,34 @@ export class LensConfigBuilder {
 
   fromAPIFormat(config: LensApiState): LensAttributes {
     const chartType = config.type;
-    if (chartType === 'metric') {
-      const converter = this.apiConvertersByChart[chartType];
-      const attributes = converter.fromAPItoLensState(config);
 
-      return {
-        ...attributes,
-        state: {
-          ...attributes.state,
-          ...filtersAndQueryToLensState(config),
-        },
-      };
+    if (!(chartType in this.apiConvertersByChart)) {
+      throw new Error(`No attributes converter found for chart type: ${chartType}`);
     }
-    throw new Error(`No attributes converter found for chart type: ${chartType}`);
+
+    const converter = this.apiConvertersByChart[chartType];
+    const attributes = converter.fromAPItoLensState(config as any); // handle type mismatches
```

Comment:
is this due to the missing charts?
Ideally we would like to get rid of `any` in the code.